packages/bigquery/src/index.js

```diff
@@ -76,6 +77,284 @@ function BigQuery(options) {
+ * @param {string=|number=} value.seconds - One or two digits (`00` - `59`).
+ * @param {string=|number=} value.fractional - Up to six digits for microsecond
+ *     precision.
+ *
+ * @example
+ * var time = bigquery.time('14:00:00'); // 2:00 PM
+ *
+ * //-
+ * // Alternatively, provide an object.
+ * //-
+ * var time = bigquery.time({
+ *   hours: 14,
+ *   minutes: 0,
+ *   seconds: 0
+ * });
+ */
+BigQuery.time =
+BigQuery.prototype.time = function(value) {
+  if (!(this instanceof BigQuery.time)) {
+    return new BigQuery.time(value);
+  }
+
+  if (is.object(value)) {
+    value = format('{h}:{m}:{s}{f}', {
+      h: value.hours,
+      m: value.minutes || 0,
+      s: value.seconds || 0,
+      f: is.defined(value.fractional) ? '.' + value.fractional : ''
+    });
+  }
+
+  this.value = value;
+};
+
+/**
+ * A timestamp represents an absolute point in time, independent of any time
+ * zone or convention such as Daylight Savings Time.
+ *
+ * @param {date} value - The time.
+ *
+ * @example
+ * var timestamp = bigquery.timestamp(new Date());
+ */
+BigQuery.timestamp =
+BigQuery.prototype.timestamp = function(value) {
+  if (!(this instanceof BigQuery.timestamp)) {
+    return new BigQuery.timestamp(value);
+  }
+
+  value = value || new Date();
+
+  if (is.date(value)) {
+    value = value.toJSON().replace(/^(.*)T(.*)Z$/, '$1 $2');
+  }
+
+  this.value = value;
+};
+
+/**
+ * Detect a value's type.
+ *
+ * @private
+ *
+ * @throws {error} If the type could not be detected.
+ *
+ * @resource [Data Type]{@link https://cloud.google.com/bigquery/data-types}
+ *
+ * @param {*} value - The value.
+ * @return {string} - The type detected from the value.
+ */
+BigQuery.getType_ = function(value) {
+  if (value instanceof BigQuery.date) {
+    return 'DATE';
+  }
+
+  if (value instanceof BigQuery.datetime) {
+    return 'DATETIME';
+  }
+
+  if (value instanceof BigQuery.time) {
+    return 'TIME';
+  }
+
+  if (value instanceof BigQuery.timestamp) {
+    return 'TIMESTAMP';
+  }
+
+  if (value instanceof Buffer) {
+    return 'BYTES';
+  }
+
+  if (is.array(value)) {
+    return 'ARRAY';
+  }
+
+  if (is.bool(value)) {
+    return 'BOOL';
+  }
+
+  if (is.number(value)) {
+    return value % 1 === 0 ? 'INT64' : 'FLOAT64';
+  }
+
+  if (is.object(value)) {
+    return 'STRUCT';
+  }
+
+  if (is.string(value)) {
+    return 'STRING';
+  }
+
+  throw new Error([
+    'This value could not be translated to a BigQuery data type.',
+    value
+  ].join('\n'));
+};
+
+/**
+ * Convert a value into a `queryParameter` object.
+ *
+ * @private
+ *
+ * @resource [Jobs.query API Reference Docs (see `queryParameters`)]{@link https://cloud.google.com/bigquery/docs/reference/rest/v2/jobs/query#request-body}
+ *
+ * @param {*} value - The value.
+ * @return {object} - A properly-formed `queryParameter` object.
+ */
+BigQuery.valueToQueryParameter_ = function(value) {
+  if (is.date(value)) {
+    value = this.timestamp(value);
```

Comment:
Getting a context error in one of the system tests here, maybe just do `BigQuery.timestamp()` to be safe?